internal/server/repository_controller.go

```diff
@@ -76,3 +70,48 @@ func (r *Repository) UpsertController(ctx context.Context, controller *Controlle
 
 	return int(rowsUpdated), nil
 }
+
+func (r *Repository) UpdateController(ctx context.Context, controller *Controller) (int, error) {
```

Comment:
I think we should make this method something more domain specific instead of a generic "update" method. Perhaps "UpdateControllerStatus"? This would help clarify that it has a particular purpose, and is the reason the `update_time` is handled differently compared to other "update" repository methods.

Please also include a go doc for the exported method.